go.mod

```diff
@@ -22,7 +22,8 @@ require (
 	github.com/openstack-k8s-operators/lib-common/modules/test v0.3.1-0.20240417144545-d24e7a32d33b
 	github.com/openstack-k8s-operators/openstack-ansibleee-operator/api v0.3.1-0.20240417073905-8802c89ece05
 	github.com/openstack-k8s-operators/openstack-baremetal-operator/api v0.3.1-0.20240417104943-679bfaa3afc1
-	golang.org/x/exp v0.0.0-20240213143201-ec583247a57a
+	github.com/openstack-k8s-operators/openstack-operator/apis v0.0.0-20240418083003-41031f2a1340
```

Comment:
This will cause a dependency loop because we have the /api and the operator in the same repo and we use git commit based pseudoversions.

1. bump something in openstack-operator/api
2. dataplane-operator bumps to the new openstack-operator/api, this results in a new pseudoversion of dataplane-operator/api as well as they are in the same repo
3. now there is a need to bump openstack-operator to the new dataplane-operator/api version, but this will lead to a new pseudoversion of openstack-operator/api as well and you go to step 2 again
I afraid this will lead to pain in the future.

So if we want to keep the logic that the dataplane-operator reads the OpenStackVersion but want to break the cycle then we either need to

* stop using pseudoversions
* move the /api module to a separate repo
* move the openstack-operator and dataplane-operator into the same repo and use replace lines in the go.mod files to use the dependency from locally and not from github (like we do in all the main go.mod files towards the /api module already)

Reply:
+1 to move to the seame repo